core/server/api_container/server/startosis_engine/kurtosis_instruction/set_service/set_service.go

```diff
@@ -0,0 +1,247 @@
+						}
+						return nil
+					},
+				},
+			},
+			Deprecation: nil,
+		},
+		Capabilities: func() kurtosis_plan_instruction.KurtosisPlanInstructionCapabilities {
+			return &SetServiceCapabilities{
+				interpretationTimeStore: interpretationTimeStore,
+				serviceNetwork:          serviceNetwork,
+				serviceName:             "",  // populated at interpretation time
+				serviceConfig:           nil, // populated at interpretation time
+				imageVal:                nil, // populated at interpretation time
+				packageId:               packageId,
+				packageContentProvider:  packageContentProvider,
+				packageReplaceOptions:   packageReplaceOptions,
+				description:             "", // populated at interpretation time
+				imageDownloadMode:       imageDownloadMode,
+			}
+		},
+		DefaultDisplayArguments: map[string]bool{
+			ServiceNameArgName: true,
+		},
+	}
+}
+
+type SetServiceCapabilities struct {
+	interpretationTimeStore *interpretation_time_value_store.InterpretationTimeValueStore
+	serviceName             service.ServiceName
+	serviceConfig           *service.ServiceConfig
+
+	serviceNetwork service_network.ServiceNetwork
+
+	// These params are needed to successfully convert service config if an ImageBuildSpec was provided
+	packageId              string
+	packageContentProvider startosis_packages.PackageContentProvider
+	packageReplaceOptions  map[string]string
+	imageVal               starlark.Value
+
+	imageDownloadMode image_download_mode.ImageDownloadMode
+	description       string
+}
+
+func (builtin *SetServiceCapabilities) Interpret(locatorOfModuleInWhichThisBuiltInIsBeingCalled string, arguments *builtin_argument.ArgumentValuesSet) (starlark.Value, *startosis_errors.InterpretationError) {
+	serviceNameArgumentValue, err := builtin_argument.ExtractArgumentValue[starlark.String](arguments, ServiceNameArgName)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", ServiceNameArgName)
+	}
+	serviceName := service.ServiceName(serviceNameArgumentValue.GoString())
+
+	builtin.serviceName = serviceName
+
+	serviceConfigOverride, err := builtin_argument.ExtractArgumentValue[*service_config.ServiceConfig](arguments, SetServiceConfigArgName)
+	if err != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract value for '%s' argument", SetServiceConfigArgName)
+	}
+	rawImageVal, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.Value](serviceConfigOverride.KurtosisValueTypeDefault, service_config.ImageAttr)
+	if interpretationErr != nil {
+		return nil, startosis_errors.WrapWithInterpretationError(err, "Unable to extract raw image attribute.")
+	}
+	if !found {
+		return nil, startosis_errors.NewInterpretationError("Unable to extract image attribute off of service config.")
+	}
+	builtin.imageVal = rawImageVal
```

Comment:
just curious - how does this work with pulling of images; say amd64 on arch64, is the fetch logic the same?

Reply:
yep, fetch logic remains the same